### PR TITLE
Difficulty cleanup

### PR DESCRIPTION
**Pull request by teor2345**
This branch contains a bunch of difficulty (and time) cleanups, which don't change any validation rules.

### CODE REVIEW

**Comment by teor2345** [comment]
This PR can't be merged until we merge #1051, because it changes some of the same code.

File: zebra-chain/src/block/header.rs [hunk 1]
```diff
@@ -67,17 +68,40 @@ pub struct Header {
     pub solution: Solution,
 }
 
+/// TODO: Inline this error into zebra_consensus::error::BlockError.
+/// See https://github.com/ZcashFoundation/zebra/issues/1021 for more details.
+#[derive(Error, Debug)]
+pub enum BlockTimeError {
+    #[error("invalid time {0:?} in block header {1:?} {2:?}: must be less than 2 hours in the future ({3:?}). Hint: check your machine's date, time, and time zone.")]
```
Comment (yaahc) [inline zebra-chain/src/block/header.rs]:
```suggestion
    #[error("invalid time {0:?} in block header {1:?} {2:?}: block time is more than 2 hours in the future ({3:?}). Hint: check your machine's date, time, and time zone.")]
```

File: zebra-chain/src/block/header.rs [hunk 2]
```diff
@@ -67,17 +68,40 @@ pub struct Header {
     pub solution: Solution,
 }
 
+/// TODO: Inline this error into zebra_consensus::error::BlockError.
```
Comment (yaahc) [inline zebra-chain/src/block/header.rs]: My intuition is that we might not want to inline this, and instead take the first error message defined here, "invalid time in block header" and make that the error message on the variant in `BlockError`. Then we make `BlockTimeError` `#[source]` for `BlockError::Time` and have it only print "must be less than 2 hours in the future {0:?}" as its error message. This way we can let the reporter split the msg into multiple lines if it wants to which will make our error reports nicer imo.

The hint bit should _really_ be handled by the Generic Member Access RFC and things like this always make me sad that this isn't a feature we have yet in rust :(((, for now though I think shoving it in the error message like you did here is the best option.

File: zebra-consensus/src/chain.rs [hunk 3]
```diff
@@ -77,7 +77,7 @@ where
 
     fn call(&mut self, block: Arc<Block>) -> Self::Future {
         let height = block.coinbase_height();
-        let span = tracing::info_span!("chain_call", ?height);
+        let span = tracing::info_span!("ChainVerifier.call", ?height);
```
Comment (yaahc) [inline zebra-consensus/src/chain.rs]: tioli:

```suggestion
        let span = tracing::info_span!("ChainVerifier::call", ?height);
```

**Review by yaahc — DISMISSED** [review]
Looks good, just some minor suggestions

File: zebra-consensus/src/block.rs [hunk 4]
```diff
@@ -99,6 +99,10 @@ where
         async move {
             let hash = block.hash();
 
+            // The height is already included in the ChainVerifier span
+            let span = tracing::debug_span!("BlockVerifier.call", ?hash);
```
Comment (teor2345) [inline zebra-consensus/src/block.rs]: Apply the same suggestion here
```suggestion
            let span = tracing::debug_span!("BlockVerifier::call", ?hash);
```

File: zebra-chain/src/block/header.rs (hunk 2 above)
Comment (teor2345) [reply zebra-chain/src/block/header.rs]:
```suggestion
/// TODO: Use this error as the source for zebra_consensus::error::BlockError::Time,
/// and make `BlockError::Time` add additional context.
```

Comment (teor2345) [reply zebra-chain/src/block/header.rs]: I moved the details to the follow-up ticket #1021.

**Comment by teor2345** [comment]
Unblocked by rebasing after #1051 merged.

Also added some cleanups for #1051.

**Review by yaahc — APPROVED** [review]

**Pull request merged**